adapters/tlsio_mbedtls.c

```diff
@@ -367,7 +387,19 @@ static int on_io_send(void *context, const unsigned char *buf, size_t sz)
     else
     {
         TLS_IO_INSTANCE *tls_io_instance = (TLS_IO_INSTANCE *)context;
-        if (xio_send(tls_io_instance->socket_io, buf, sz, on_send_complete, tls_io_instance) != 0)
+        ON_SEND_COMPLETE on_complete_callback = NULL;
+        void *context = NULL;
+
+        // Only allow Application data type message to send on_send_complete callback.
+        // In case when sending more then max_fregment size trigger only on when last data
```

Comment:
nit: "then" should be "than" and "max_fregment" -> "max_fragment"

Reply:
Ack

---
In reply to: [354466478](https://github.com/Azure/azure-c-shared-utility/pull/421#discussion_r354466478) [](ancestors = 354466478)